### ISSUE TITLE
Video never plays when using remote transcoding, despite worker logs

### ISSUE THREAD

**Issue by EvanBGoldstein** [pabloromeo/clusterplex#132]
Remote transcoding appears to be working (judging by the logs on the worker), but video never plays. The player (both in app and browser) just sits there spinning.

If I switch to local transcoding only, it plays just fine.

Logs from worker:
```
[AVIOContext @ 0x7f77319999c0] Statistics: 0 seeks, 1 writeouts
[segment @ 0x7f7736ba8040] segment:'chunk-00474' starts with packet stream:0 pts:2518133 pts_time:2518.13 frame:75544
*** 1 dup!
    Last message repeated 43 times
[segment @ 0x7f7736ba8040] Opening 'http://mediaserver:32400/video/:/transcode/session/658qje7co7d2xuiq4syamcso/dfadd804-7aed-407e-9556-c7fa5618b665/manifest?X-Plex-Http-Pipeline=infinite' for writing
[tcp @ 0x7f7734f4b100] Starting connection attempt to 172.16.50.119 port 32400
[tcp @ 0x7f7734f4b100] Successfully connected to 172.16.50.119 port 32400
[AVIOContext @ 0x7f7736596680] Statistics: 0 seeks, 1 writeouts
[segment @ 0x7f7736ba8040] segment:'chunk-00474' count:474 ended
[AVIOContext @ 0x7f773c7e3ec0] Statistics: 0 seeks, 2 writeouts
[segment @ 0x7f7736ba8040] Opening 'chunk-00475' for writing
[segment @ 0x7f7736ba8040] Opening 'http://mediaserver:32400/video/:/transcode/session/658qje7co7d2xuiq4syamcso/dfadd804-7aed-407e-9556-c7fa5618b665/manifest?X-Plex-Http-Pipeline=infinite' for writing
[tcp @ 0x7f7734f4b100] Starting connection attempt to 172.16.50.119 port 32400
[tcp @ 0x7f7734f4b100] Successfully connected to 172.16.50.119 port 32400
[AVIOContext @ 0x7f7731999600] Statistics: 0 seeks, 1 writeouts
[segment @ 0x7f7736ba8040] segment:'chunk-00475' starts with packet stream:0 pts:2526467 pts_time:2526.47 frame:75794
*** 1 dup!
    Last message repeated 16 times
[segment @ 0x7f7736ba8040] Opening 'http://mediaserver:32400/video/:/transcode/session/658qje7co7d2xuiq4syamcso/dfadd804-7aed-407e-9556-c7fa5618b665/manifest?X-Plex-Http-Pipeline=infinite' for writing
[tcp @ 0x7f7734f4b100] Starting connection attempt to 172.16.50.119 port 32400
[tcp @ 0x7f7734f4b100] Successfully connected to 172.16.50.119 port 32400
[AVIOContext @ 0x7f7731998a00] Statistics: 0 seeks, 1 writeouts
[segment @ 0x7f7736ba8040] segment:'chunk-00475' count:475 ended
[AVIOContext @ 0x7f773c7e3ec0] Statistics: 0 seeks, 1 writeouts
[segment @ 0x7f7736ba8040] Opening 'chunk-00476' for writing
[segment @ 0x7f7736ba8040] Opening 'http://mediaserver:32400/video/:/transcode/session/658qje7co7d2xuiq4syamcso/dfadd804-7aed-407e-9556-c7fa5618b665/manifest?X-Plex-Http-Pipeline=infinite' for writing
[tcp @ 0x7f7734f4b100] Starting connection attempt to 172.16.50.119 port 32400
[tcp @ 0x7f7734f4b100] Successfully connected to 172.16.50.119 port 32400
[AVIOContext @ 0x7f7736596680] Statistics: 0 seeks, 1 writeouts
[segment @ 0x7f7736ba8040] segment:'chunk-00476' starts with packet stream:0 pts:2528967 pts_time:2528.97 frame:75869
*** 1 dup!
    Last message repeated 42 times
[segment @ 0x7f7736ba8040] Opening 'http://mediaserver:32400/video/:/transcode/session/658qje7co7d2xuiq4syamcso/dfadd804-7aed-407e-9556-c7fa5618b665/manifest?X-Plex-Http-Pipeline=infinite' for writing
[tcp @ 0x7f7734f4b100] Starting connection attempt to 172.16.50.119 port 32400
[tcp @ 0x7f7734f4b100] Successfully connected to 172.16.50.119 port 32400
[AVIOContext @ 0x7f7731999dc0] Statistics: 0 seeks, 1 writeouts
[segment @ 0x7f7736ba8040] segment:'chunk-00476' count:476 ended
[AVIOContext @ 0x7f773c7e3ec0] Statistics: 0 seeks, 3 writeouts
[segment @ 0x7f7736ba8040] Opening 'chunk-00477' for writing
[segment @ 0x7f7736ba8040] Opening 'http://mediaserver:32400/video/:/transcode/session/658qje7co7d2xuiq4syamcso/dfadd804-7aed-407e-9556-c7fa5618b665/manifest?X-Plex-Http-Pipeline=infinite' for writing
[tcp @ 0x7f7734f4b100] Starting connection attempt to 172.16.50.119 port 32400
[tcp @ 0x7f7734f4b100] Successfully connected to 172.16.50.119 port 32400
[AVIOContext @ 0x7f7731998a00] Statistics: 0 seeks, 1 writeouts
[segment @ 0x7f7736ba8040] segment:'chunk-00477' starts with packet stream:0 pts:2537300 pts_time:2537.3 frame:76119
*** 1 dup!
    Last message repeated 31 times
*** 2 dup!
*** 1 dup!
    Last message repeated 9 times
[segment @ 0x7f7736ba8040] Opening 'http://mediaserver:32400/video/:/transcode/session/658qje7co7d2xuiq4syamcso/dfadd804-7aed-407e-9556-c7fa5618b665/manifest?X-Plex-Http-Pipeline=infinite' for writing
[tcp @ 0x7f7734f4b100] Starting connection attempt to 172.16.50.119 port 32400
[tcp @ 0x7f7734f4b100] Successfully connected to 172.16.50.119 port 32400
[AVIOContext @ 0x7f7736596680] Statistics: 0 seeks, 1 writeouts
[segment @ 0x7f7736ba8040] segment:'chunk-00477' count:477 ended
[AVIOContext @ 0x7f773c7e3ec0] Statistics: 0 seeks, 3 writeouts
[segment @ 0x7f7736ba8040] Opening 'chunk-00478' for writing
[segment @ 0x7f7736ba8040] Opening 'http://mediaserver:32400/video/:/transcode/session/658qje7co7d2xuiq4syamcso/dfadd804-7aed-407e-9556-c7fa5618b665/manifest?X-Plex-Http-Pipeline=infinite' for writing
[tcp @ 0x7f7734f4b100] Starting connection attempt to 172.16.50.119 port 32400
[tcp @ 0x7f7734f4b100] Successfully connected to 172.16.50.119 port 32400
[AVIOContext @ 0x7f7731999600] Statistics: 0 seeks, 1 writeouts
[segment @ 0x7f7736ba8040] segment:'chunk-00478' starts with packet stream:0 pts:2545633 pts_time:2545.63 frame:76369
*** 1 dup!
Killing child process for task 878ccd44-3ed3-4b1f-944b-8f4264adeb00
Removing process from taskMap
    Last message repeated 20 times
[segment @ 0x7f7736ba8040] Opening 'http://mediaserver:32400/video/:/transcode/session/658qje7co7d2xuiq4syamcso/dfadd804-7aed-407e-9556-c7fa5618b665/manifest?X-Plex-Http-Pipeline=infinite' for writing
[tcp @ 0x7f7734f4b100] Starting connection attempt to 172.16.50.119 port 32400
[tcp @ 0x7f7734f4b100] Successfully connected to 172.16.50.119 port 32400
[AVIOContext @ 0x7f77319988c0] Statistics: 0 seeks, 1 writeouts
[segment @ 0x7f7736ba8040] segment:'chunk-00478' count:478 ended
[AVIOContext @ 0x7f773c7e3ec0] Statistics: 0 seeks, 2 writeouts
frame=76524 fps=159 q=-1.0 Lsize=N/A time=00:42:29.69 bitrate=N/A dup=13576 drop=0 speed=5.31x    
video:157307kB audio:47330kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
Input file #0 (/mnt/Media/Documentaries/Four Blood Moons (2015)/Four Blood Moons (2015).mkv):
  Input stream #0:0 (video): 62956 packets read (510708041 bytes); 62949 frames decoded; 
  Input stream #0:1 (audio): 109870 packets read (48465734 bytes); 
  Total: 172826 packets (559173775 bytes) demuxed
Output file #0 (chunk-%05d):
  Output stream #0:0 (video): 76524 frames encoded; 76524 packets muxed (161081877 bytes); 
  Output stream #0:1 (audio): 109870 packets muxed (48465734 bytes); 
  Total: 186394 packets (209547611 bytes) muxed
[libx264 @ 0x7f773ad9d600] frame I:541   Avg QP:16.82  size: 24965
[libx264 @ 0x7f773ad9d600] frame P:23295 Avg QP:19.84  size:  4766
[libx264 @ 0x7f773ad9d600] frame B:52688 Avg QP:22.95  size:   694
[libx264 @ 0x7f773ad9d600] consecutive B-frames:  2.9% 13.2%  7.7% 76.1%
[libx264 @ 0x7f773ad9d600] mb I  I16..4: 21.6% 28.2% 50.1%
[libx264 @ 0x7f773ad9d600] mb P  I16..4:  4.2%  6.1%  2.1%  P16..4: 24.6% 11.6%  7.1%  0.0%  0.0%    skip:44.3%
[libx264 @ 0x7f773ad9d600] mb B  I16..4:  0.4%  0.6%  0.1%  B16..8:  7.9%  2.9%  0.4%  direct: 2.5%  skip:85.2%  L0:39.6% L1:39.9% BI:20.5%
[libx264 @ 0x7f773ad9d600] 8x8 transform intra:46.9% inter:32.5%
[libx264 @ 0x7f773ad9d600] coded y,uvDC,uvAC intra: 53.1% 46.9% 15.7% inter: 5.9% 6.3% 0.2%
[libx264 @ 0x7f773ad9d600] i16 v,h,dc,p: 51% 24% 17%  8%
[libx264 @ 0x7f773ad9d600] i8 v,h,dc,ddl,ddr,vr,hd,vl,hu: 29% 21% 19%  4%  5%  6%  5%  5%  6%
[libx264 @ 0x7f773ad9d600] i4 v,h,dc,ddl,ddr,vr,hd,vl,hu: 28% 20% 11%  6%  7%  8%  7%  7%  6%
[libx264 @ 0x7f773ad9d600] i8c dc,h,v,p: 59% 16% 20%  5%
[libx264 @ 0x7f773ad9d600] Weighted P-Frames: Y:5.4% UV:1.8%
[libx264 @ 0x7f773ad9d600] kb/s:505.19
[AVIOContext @ 0x7f773c7e3c40] Statistics: 560090694 bytes read, 2 seeks
Exiting normally, received signal 15.
Completed transcode
Removing process from taskMap
```

**Comment by pabloromeo** [comment]
Yeah, the worker log look correct. 
It would appear that Plex is not seeing the new chunks being transcoded in the shared transcode directory. There might be relevant info in Plex's own logs that might shed some light on what's going on. You might have to turn on debug logs in Plex's config and view them from the Web Console.

Are you sure that same exact path is available both to PMS and to the Worker? Have you also configured the Network settings in PMS to cover the subnets of the workers? (https://github.com/pabloromeo/clusterplex#network-settings-in-pms)

**Comment by EvanBGoldstein** [comment]
The same directory is mounted to /tmp on both containers. Plex is configured to use /tmp/transcoding for the transcoder temporary directory. As long as /tmp is shared between the containers, there isn't anything I need to do to tell the worker to use that directory, right?

I enabled debug logging in Plex's general settings, and did not see anything in the logs that wasn't there before. Here are the logs once the transcoding starts:

```
Calling external transcoder: /app/transcoder.js
args => ["-codec:0","h264","-codec:1","dca","-analyzeduration","20000000","-probesize","20000000","-i","/mnt/Media/Documentaries/Apollo 11 (2019)/Apollo 11 (2019) [DTS-HD MA 5.1].mkv","-filter_complex","[0:0]scale=w=1920:h=872:force_divisible_by=4[0];[0]format=pix_fmts=yuv420p|nv12[1]","-map","[1]","-metadata:s:0","language=eng","-codec:0","libx264","-crf:0","20","-maxrate:0","11178k","-bufsize:0","22356k","-r:0","23.975999999999999","-preset:0","veryfast","-x264opts:0","subme=0:me_range=4:rc_lookahead=10:me=hex:8x8dct=0:partitions=none","-force_key_frames:0","expr:gte(t,n_forced*3)","-filter_complex","[0:1] aresample=async=1:ocl='stereo':rematrix_maxval=0.000000dB:osr=48000[2]","-map","[2]","-metadata:s:1","language=eng","-codec:1","aac","-b:1","180k","-f","dash","-seg_duration","3","-dash_segment_type","mp4","-init_seg_name","init-stream$RepresentationID$.m4s","-media_seg_name","chunk-stream$RepresentationID$-$Number%05d$.m4s","-window_size","5","-delete_removed","false","-skip_to_segment","1","-time_delta","0.0625","-manifest_name","http://mediaserver:32400/video/:/transcode/session/x9h3f3qktchlbnzkx1xgetk0/0527b209-6062-409f-8c68-605d46e8b8ed/manifest?X-Plex-Http-Pipeline=infinite","-avoid_negative_ts","disabled","-map_metadata","-1","-map_chapters","-1","dash","-start_at_zero","-copyts","-vsync","cfr","-y","-nostats","-loglevel","verbose","-loglevel_plex","verbose","-progressurl","http://mediaserver:32400/video/:/transcode/session/x9h3f3qktchlbnzkx1xgetk0/0527b209-6062-409f-8c68-605d46e8b8ed/progress"]
cwd => "/tmp/transcoding/Transcode/Sessions/plex-transcode-x9h3f3qktchlbnzkx1xgetk0-0527b209-6062-409f-8c68-605d46e8b8ed"
env => {"PUID":"21114","PLEX_ARCH":"amd64","HOSTNAME":"dd253f78818d","LANGUAGE":"en_US.UTF-8","TRANSCODE_OPERATING_MODE":"remote","ORCHESTRATOR_URL":"http://orchestrator:3500","PWD":"/tmp/transcoding/Transcode/Sessions/plex-transcode-x9h3f3qktchlbnzkx1xgetk0-0527b209-6062-409f-8c68-605d46e8b8ed","PLEX_DOWNLOAD":"https://downloads.plex.tv/plex-media-server-new","PLEX_MEDIA_SERVER_MAX_PLUGIN_PROCS":"6","NVIDIA_DRIVER_CAPABILITIES":"compute,video,utility","PMS_IP":"mediaserver","TZ":"America/New_York","PLEX_MEDIA_SERVER_USER":"abc","HOME":"/root","LANG":"en_US.UTF-8","PGID":"21114","TERM":"xterm","PLEX_MEDIA_SERVER_INFO_VENDOR":"Docker","PLEX_MEDIA_SERVER_HOME":"/usr/lib/plexmediaserver","DOCKER_MODS":"ghcr.io/pabloromeo/clusterplex_dockermod:latest","X_PLEX_TOKEN":"LFUsixjShAdXHxL6vVJf","PLEX_MEDIA_SERVER_INFO_MODEL":"x86_64","SHLVL":"0","LD_LIBRARY_PATH":"/usr/lib","PLEX_MEDIA_SERVER_INFO_PLATFORM_VERSION":"4.15.0-169-generic","LIBVA_DRIVERS_PATH":"/usr/lib/plexmediaserver/lib/dri","PLEX_MEDIA_SERVER_APPLICATION_SUPPORT_DIR":"/config/Library/Application Support","CWD":"/","PATH":"/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin","VERSION":"docker","DEBIAN_FRONTEND":"noninteractive","PLEX_MEDIA_SERVER_INFO_DEVICE":"Docker Container (LinuxServer.io)","FFMPEG_EXTERNAL_LIBS":"/config/Library/Application\\ Support/Plex\\ Media\\ Server/Codecs/994f4ee-4285-linux-x86_64/","TRANSCODER_VERBOSE":"1"}
Orchestrator requesting pending work
JobPoster connected, announcing
Sending request to orchestrator on: http://orchestrator:3500
```

Here is the docker compose:

```
services:
  mediaserver:
    image: ghcr.io/linuxserver/plex:latest
    #user: '21114:21114'
    environment:
      DOCKER_MODS: "ghcr.io/pabloromeo/clusterplex_dockermod:latest"
      VERSION: docker
      PUID: 21114
      PGID: 21114
      TZ: America/New_York
      ORCHESTRATOR_URL: http://orchestrator:3500
      PMS_IP: mediaserver
      TRANSCODE_OPERATING_MODE: remote #(local|remote|both)
      TRANSCODER_VERBOSE: "1"   # 1=verbose, 0=silent      
    healthcheck:
      test: curl -fsS http://localhost:32400/identity > /dev/null || exit 1
      interval: 15s
      timeout: 15s
      retries: 5
      start_period: 120s
    volumes:
      - /mnt/ApplicationData/plex/config:/config
      - /mnt/Apps-nfs/plex/backups:/backups
      - /mnt/Media:/mnt/Media
      - /mnt/Apps-nfs/plex/tmp:/tmp
      - /etc/localtime:/etc/localtime:ro
    networks:
      - plex
      - traefik
    ports:
      - 32469:32469
      - 32400:32400
      - 3005:3005
      - 8324:8324
      - 1900:1900/udp
      - 32410:32410/udp
      - 32412:32412/udp
      - 32413:32413/udp
      - 32414:32414/udp
    deploy:
      mode: replicated
      replicas: 1
      labels:
        - traefik.enable=true
        - traefik.http.routers.plex.entrypoints=https
        - traefik.http.routers.plex.rule=Host(`plex.app.icarusnet.me`)
        - traefik.http.routers.plex.tls=true
        - traefik.http.services.plex.loadbalancer.server.port=32400
        
  orchestrator:
    image: ghcr.io/pabloromeo/clusterplex_orchestrator:latest
    deploy:
      mode: replicated
      replicas: 1
      update_config:
        order: start-first
    healthcheck:
      test: curl -fsS http://localhost:3500/health > /dev/null || exit 1
      interval: 15s
      timeout: 15s
      retries: 5
      start_period: 30s
    environment:
      TZ: America/New_York
      STREAM_SPLITTING: "OFF" # ON | OFF (default)
      LISTENING_PORT: 3500
      EAE_ROOT: /tmp/Transcode/eae_root
      WORKER_SELECTION_STRATEGY: "LOAD_RANK" # RR | LOAD_CPU | LOAD_TASKS | LOAD_RANK (default)
    volumes:
      - /etc/localtime:/etc/localtime:ro
    networks:
      - plex
    ports:
      - 3500:3500

  worker1:
    image: ghcr.io/linuxserver/plex:latest
    #image: ghcr.io/pabloromeo/clusterplex_worker:latest
    #user: '21114:21114'
    hostname: "plex-worker-{{.Node.Hostname}}"
    deploy:
      #mode: global
      update_config:
        order: start-first
    environment:
      DOCKER_MODS: "ghcr.io/pabloromeo/clusterplex_worker_dockermod:latest"
      VERSION: docker
      #EAE_ROOT: /tmp/Transcode/eae_root
      PUID: 21114
      PGID: 21114
      TZ: America/New_York
      LISTENING_PORT: 3501      # used by the healthcheck
      STAT_CPU_INTERVAL: 2000   # interval for reporting worker load metrics
      TRANSCODER_VERBOSE: "1"   # 1=verbose, 0=silent      

      ORCHESTRATOR_URL: http://orchestrator:3500
    healthcheck:
      test: curl -fsS http://localhost:3501/health > /dev/null || exit 1
      interval: 15s
      timeout: 15s
      retries: 5
      start_period: 240s
    volumes:
      - /mnt/Apps-nfs/plex/codecs:/codecs # (optional, can be used to share codecs)
      - /mnt/Media:/mnt/Media
      - /mnt/Apps-nfs/plex/tmp:/tmp
      - /etc/localtime:/etc/localtime:ro
    networks:
      - plex
```


I appreciate the help!

**Comment by pabloromeo** [comment]
Did you review Plex's network settings section to allow access from docker's subnet? Others have found that to be necessary. Without it Plex would ignore what the worker was doing.

**Comment by EvanBGoldstein** [comment]
I believe I've solved it. The network settings were correct, and plex's web console confirmed the IP address of the worker was being allowed without auth, but I noticed the message `Request came in with unrecognized domain / IP 'mediaserver' in header Host; treating as non-local`. I updated the docker compose to set the hostname of the plex container to be the same hostname that was set using the `PMS_IP` environment variable, and now it plays.

Thanks again for your help!
